File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/view_models/RepositoriesViewModel.java
package com.mondiamedia.ahmedbadr.githubreopos.view_models;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.mondiamedia.ahmedbadr.githubreopos.models.GitRepo;
import com.mondiamedia.ahmedbadr.githubreopos.repository.DataRepository;

import java.util.List;

import javax.inject.Inject;

public class RepositoriesViewModel extends ViewModel {

    private DataRepository dataRepository;
    private MutableLiveData<List<GitRepo>> mReposList;

    @Inject
    public RepositoriesViewModel(DataRepository dataRepository) {
        this.dataRepository = dataRepository;
    }

    public void init() {
        if (mReposList != null) {
            return;
        }

        mReposList = dataRepository.getRepos();
    }

    public LiveData<List<GitRepo>> getRepos() {
        return mReposList;
    }

    public LiveData<List<GitRepo>> refreshRepos() {
        mReposList = dataRepository.getRefreshRepos();
        return mReposList;
    }
}

File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/view_models/RepositoriesViewModel.kt
package com.mondiamedia.ahmedbadr.githubreopos.view_models

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.mondiamedia.ahmedbadr.githubreopos.models.GitRepo
import com.mondiamedia.ahmedbadr.githubreopos.repository.DataRepository
import kotlinx.coroutines.launch
import javax.inject.Inject

class RepositoriesViewModel @Inject
constructor(private val dataRepository: DataRepository) : ViewModel() {
    private var _reposList = MutableLiveData<List<GitRepo>>()
    val reposList: LiveData<List<GitRepo>>?
        get() = _reposList

    fun init() {
        viewModelScope.launch {
            reposList?.value.let {
                if (!it.isNullOrEmpty()) return@launch
            }

            _reposList.value = dataRepository.getRepos()
        }
    }

    suspend fun refreshRepos() {
        _reposList.value = dataRepository.getRefreshRepos()
    }
}
